File: src/main/java/ca/usherbrooke/fgen/api/Entities/WatchMovieUser.java
package ca.usherbrooke.fgen.api.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "watch_movie_user", schema = "app")
public class WatchMovieUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    public User user;

    @ManyToOne
    @JoinColumn(name = "movie_id", nullable = false)
    @JsonIgnore
    public Movie movie;

    @Enumerated(EnumType.STRING)
    public MovieStatus status;

    public boolean saved;
    public Integer rating;
}

File: moviestreamer-ui/src/entities/WatchMovieUser.js

export const MovieStatus = Object.freeze({
    WATCHED: 'WATCHED',
    WATCHING: 'WATCHING',
    PLAN_TO_WATCH: 'PLAN_TO_WATCH'
})

export class WatchMovieUser {
    constructor({ id, status, saved, rating } = {}) {
        this.id = id
        this.status = status   // one of MovieStatus values
        this.saved = saved
        this.rating = rating
        // user and movie are @JsonIgnore on the backend, so they won't appear here
    }
}